### PR TITLE
Add DailyReward2, DailyRewardMail

### PR DESCRIPTION
**Pull request by hskim881028**
To add a daily reward item(`400000,모래시계,Hourglass,4,Normal`), I did `DailyReward` version up  and created '`DailyRewardMail`

### CODE REVIEW

File: .Lib9c.Tests/Action/DailyRewardTest2.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            Assert.Throws<FailedLoadStateException>(() => action.Execute(new ActionContext()
+                {
+                    PreviousStates = new State(),
+                    Signer = _agentAddress,
+                    BlockIndex = 0,
+                })
+            );
+        }
+    }
+}
```
Comment (ipdae) [inline .Lib9c.Tests/Action/DailyRewardTest2.cs:109]:
```suggestion
}

```

File: Lib9c/Model/Mail/DailyRewardMail.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            : base(serialized)
+        {
+        }
+
+        public override void Read(IMail mail)
+        {
+            mail.Read(this);
+        }
+    }
+}
```
Comment (ipdae) [inline Lib9c/Model/Mail/DailyRewardMail.cs:28]:
```suggestion
}

```

File: Lib9c/Action/DailyReward2.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#pragma warning disable LAA1002
+                new Bencodex.Types.Dictionary(new Dictionary<IKey, IValue>
+                {
+                    [(Text) "materials"] = materials.Serialize(),
+                    [(Text) "id"] = id.Serialize(),
+                }.Union((Bencodex.Types.Dictionary) base.Serialize()));
+#pragma warning restore LAA1002
+        }
+    }
+}
```
Comment (ipdae) [inline Lib9c/Action/DailyReward2.cs:110]:
```suggestion
}

```

File: Lib9c/Action/DailyReward2.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+using Libplanet.Action;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.Mail;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType("daily_reward2")]
```
Comment (ipdae) [inline Lib9c/Action/DailyReward2.cs:16]: 기존의 `DailyReward` 코드를 `DailyReward2` 로 옮기고, `DailyReward` 클래스의 액션타입과 동작을 업데이트하는게 변경점에 대한 diff 파악과 사용하는쪽에서의 변경점이 적을것 같습니다. (RankingBattle처럼)

File: .Lib9c.Tests/Action/DailyRewardTest2.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+            _initialState = _initialState
+                .SetState(Addresses.GameConfig, new GameConfigState().Serialize())
+                .SetState(_agentAddress, agentState.Serialize())
+                .SetState(_avatarAddress, avatarState.Serialize());
+        }
+
+        [Theory]
+        [InlineData(400000, 10)]
+        public void Execute(int itemId, int itemCount)
```
Comment (moreal) [inline .Lib9c.Tests/Action/DailyRewardTest2.cs:62]: The numbers, 400000 and 10, seem like the hard-coded values in `DailyReward2`. Therefore it seems not configurable with parameters. And I have wondered why this test uses `Theory`, it seems enough to use `Fact` 🤔 ? How do you think about this? 🙏

Comment (hskim881028) [reply .Lib9c.Tests/Action/DailyRewardTest2.cs:62]: > The numbers, 400000 and 10, seem like the hard-coded values in `DailyReward2`. Therefore it seems not configurable with parameters. And I have wondered why this test uses `Theory`, it seems enough to use `Fact` 🤔 ? How do you think about this? 🙏

I used test 'theory' with the assumption that the hardcoded values ​​will be modified later.
But when I think about it again.. :thinking_face:
Now it seems correct to use 'fact' and then use 'theory' when the hardcoding is modified (probably v3).

File: Lib9c/Action/DailyReward2.cs (hunk 4 above)
Comment (hskim881028) [reply Lib9c/Action/DailyReward2.cs]: I swapped 'DailyReward' and 'DailyReward2' . please review again🙏 @ipdae

**Review by area363 — APPROVED** [review]

File: Lib9c/Action/DailyReward.cs [hunk 6]
```diff
@@ -1,18 +1,22 @@
...
 
 namespace Nekoyume.Action
 {
     [Serializable]
-    [ActionType("daily_reward")]
+    [ActionType("daily_reward2")]
     public class DailyReward : GameAction
     {
-        public Address avatarAddress;
+                public Address avatarAddress;
```
Comment (ipdae) [inline Lib9c/Action/DailyReward.cs:19]:
```suggestion
        public Address avatarAddress;
```

**Review by moreal — DISMISSED** [review]
The build seems failed because of [SA1507 Rule](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1507.md). Could you fix it please? @hskim881028

**Comment by hskim881028** [comment]
> The build seems failed because of [SA1507 Rule](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1507.md). Could you fix it please? @hskim881028

Modifications completed~!

**Review by area363 — APPROVED** [review]

**Review by ipdae — APPROVED** [review]

**Review by moreal — COMMENTED** [review]
I leave _comment_ review to remove *request changes* mark.

**Review by Unengine — APPROVED** [review]

**Pull request merged**